source/change-streams/tests/change-streams.yml

```diff
@@ -0,0 +1,197 @@
+database_name: &database_name "change-stream-tests"
+collection_name_2: &collection_name_2 "test2"
+database_name_2: &database_name_2 "change-stream-tests-2"
+defaults:
+  minServerVersion: "3.6.0"
+  topology: replicaset
+  target: collection
+  numChanges: 1
+  result: {}
+  expectations: []
+  operations:
+    -
+      database: *database_name
+      collection: *collection_name_2
+      commandName: insert
+      arguments:
+        - { x: 1 }
+    -
+      database: *database_name_2
+      collection: *collection_name
+      commandName: insert
+      arguments:
+        - { y: 2 }
+    -
+      database: *database_name
+      collection: *collection_name
+      commandName: insert
+      arguments:
+        - { z: 3 }
+tests:
+  -
+    description: "`$changeStream` must be the first stage in a change stream pipeline sent to the server"
+    expectations:
+      - 
+        command_started_event:
+          command:
+            aggregate: *collection_name
+            pipeline:
+              - {$changeStream: {}}
+          command_name: aggregate
+          database_name: *database_name
+  -
+    description: The watch helper must not throw a custom exception when executed against a single server topology, but instead depend on a server error
+    topology: single
+    operations: []
+    result:
+      error:
+        errmsg: The $changeStream stage is only supported on replica sets
+        code: 40573
+  -
+    description: ChangeStream must continuously track the last seen resumeToken
+    skip: true
+  -
+    description: ChangeStream will throw an exception if the server response is missing the resume token
+    skip: true
+  -
+    description: ChangeStream will automatically resume one time on a resumable error (including not master) with the initial pipeline and options, except for the addition/update of a resumeToken.
+    skip: true
+  -
+    description: ChangeStream will not attempt to resume on a server error
+    skip: true
+  -
+    description: ChangeStream will perform server selection before attempting to resume, using initial readPreference
+    skip: true
+  -
+    description: Ensure that a cursor returned from an aggregate command with a cursor id, and an initial empty batch, is not closed on the driver side.
+    skip: true
+  -
+    description: The killCursors command sent during the “Resume Process” must not be allowed to throw an exception.
+    skip: true
+  -
+    description: The server returns change stream responses in the specified ``ServerResponse`` format
+    result:
+      success:
+        -
+          _id: {}
+          operationType: insert
+          ns:
+            db: *database_name
+            coll: *collection_name
+          documentKey: {}
+          fullDocument:
+            z: 3
+  -
+    description: Executing a `watch` helper on a Collection results in notifications for changes to the specified collection
+    expectations:
+      - 
+        command_started_event:
```

Comment:
I'm surprised we don't have a getMore event as well, since according to the README we start the change stream (which sends this aggregate), before inserting the documents.  In order to be notified of the documents we'd have to issue a getMore

Reply:
1. From what I can tell, a `getMore` is not necessarily guaranteed, since results can be returned in the initial `find`
2. The test is more about asserting that the change stream is against a specific collection, which is why we are checking the initial aggregate.

Reply:
1. It's guaranteed in this case because the document insertion happens after the change stream has been established with the initial `aggregate` command
2. OK, but are drivers expected to ignore any getMore commands when comparing?  If so, say that in the README.  But better if we can actually assert on the getMore commands, if the behavior is predicable.

Reply:
I wrote in the README:

> - If there are any ``expectations``
>   - assert that they MATCH the actual APM events

This is meant to equate to the following pseudocode:

```
for (expected, idx) in expectations:
  Assert that actual[idx] MATCHES expected
```

If that is not clear, I can make it more elaborate.

I'd also rather not have a test for the getMore, since it is not actually relevant to the assertion